ocaml/otherlibs/dynlink/byte/dynlink.ml

```diff
@@ -196,31 +150,24 @@ module Bytecode = struct
           raise (DT.Error (Linking_error (file_name, new_error)))
         end;
         (* PR#5215: identify this code fragment by
-<<<<<<< HEAD
           digest of file contents + unit name.
           Unit name is needed for .cma files, which produce several code
           fragments. *)
+        (* XXX mshinwell: conflict resolution around here needs checking *)
         let digest =
           Digest.string
             (file_digest ^ Compilation_unit.full_path_as_string compunit.cu_name)
         in
-||||||| 121bedcfd2
-           digest of file contents + unit name.
-           Unit name is needed for .cma files, which produce several code
-           fragments. *)
-        let digest = Digest.string (file_digest ^ compunit.cu_name) in
-=======
-           digest of file contents + unit name.
-           Unit name is needed for .cma files, which produce several code
-           fragments. *)
-        let unit_name = Symtable.Compunit.name compunit.cu_name in
-        let digest = Digest.string (file_digest ^ unit_name) in
->>>>>>> 5.2.0
         let events =
           if compunit.cu_debug = 0 then [| |]
           else begin
             seek_in ic compunit.cu_debug;
-            [| (Compression.input_value ic : Instruct.debug_event list) |]
+            [|
+              (* CR ocaml 5 compressed-marshal:
+              (Compression.input_value ic : Instruct.debug_event list)
+              *)
+              (Marshal.from_channel ic : Instruct.debug_event list)
```

Comment:
How are we sure it is kept synchronized
with the code in `Emitcode`? Are the
tests enough?

Reply:
Xavier and I discussed this, it seems ok.